modules/core/src/main/scala/org/tessellation/infrastructure/trust/storage/TrustStorage.scala

```diff
@@ -53,17 +54,25 @@ object TrustStorage {
   def make[F[_]: Monad: Ref.Make](
     trustUpdates: Option[PeerObservationAdjustmentUpdateBatch],
     config: TrustStorageConfig,
-    seedlist: Option[Set[PeerId]]
+    seedlist: Option[Set[SeedlistEntry]]
   ): F[TrustStorage[F]] =
     trustUpdates.fold(make(config, seedlist))(batch => make(config, seedlist).flatTap(_.updateTrust(batch)))
 
   def make[F[_]: Monad](
     trustStoreRef: Ref[F, TrustStore],
     config: TrustStorageConfig,
-    seedlist: Option[Set[PeerId]]
+    seedlist: Option[Set[SeedlistEntry]]
   ): TrustStorage[F] =
     new TrustStorage[F] {
 
+      val seedlistScores: Option[Map[PeerId, TrustValueRefined]] =
+        seedlist
+          .map(
+            _.filter(_.bias.exists(_.value > 0.0))
+              .map(e => e.peerId -> e.bias.getOrElse(defaultPeerTrustScore))
```

Comment:
That `getOrElse` never gets called because you have already filtered all the `None` in line `71`.

If you have such seedlist entries:

```
A 0.0
B 1.0
C -1.0
```

then your code makes such Map:

```
Map(B -> 1.0)
```

If you want to keep values which are `<= 0.0` but just default them to `defaultPeerTrustScore` then you should map over entries and convert all the `<=` to `defaultPeerTrustScore`:

```scala
seedlist.map(
  _.map(_.bias.filter(_.value > 0.0).getOrElse(defaultPeerTrustScore))
)
```

Reply:
Changed to

```
        seedlist
          .map(
            _.map { e =>
              e.peerId -> e.bias.filter(_.value > 0.0).getOrElse(defaultPeerTrustScore)
            }.toMap
          )
```

Reply:
Doesn't the `getOrElse` mean that the lowest "score" someone can have is the default value? In other words, if the bias is determined to be the lowest value (-1?) then it automatically gets upgraded to the default. Is that what we want?

Reply:
Right now we're not really using any seed list entry to denote a negative score (i.e. the equivalent of downranking a node due to negative behavior.) The getOrElse here should be supplying a value for all entries on the seed list that otherwise have no score, so the interpretation of this is that we haven't updated old entries in the seed list to have specified scores, so if no information is present they should default to a preset default.

It might be useful to include negative scores in the future (for some sort of slasher like mechanism,) but for now it's not used at all. Still, if a score WAS negative, this function would be incorrect, because we'd just be filling it with a default.

The correct behavior would be to disregard a negative score, take the value from the seedlist if filled, otherwise default to constant.

Reply:
Changed to

```
        seedlist
          .map(
            _.map { e =>
              e.peerId -> e.bias
            }.collect { case (peerId, Some(s)) if s.value > 0.0 => peerId -> s }.toMap
          )
```